### DISCUSSION TITLE
What does the cmp source do?

### DISCUSSION THREAD

**Discussion by max397574** [folke/lazydev.nvim#52] [General]
What exactly does the cmp source do? If luals is configured to do get completions from the various plugins or repos why is an additional source needed then? Can't all be done with a simple lsp-source?

Asking because this could be relevant for me cause I'm currently writing a new completion engine (ps @folke asked you something on matrix about that)

**Comment by folke** [comment]
It's not 100% needed, but it provides completion for `require("foo...")` and `---@module "foo..."`.
Since the plugins are not yet part of the workspace, luals doesn't know about them and can't provide completions for these
require calls.

The lazydev implementation is also a lot faster for some reason.

Looking forward to that completion engine!

**Reply by max397574** [reply]
makes sense I think
Have to play around with it a little bit until I can give better feedback

**Reply by max397574** [reply]
Another question I had
How (if at all) does lazydev integrate with neoconf?
Can you change the settings per project?
would be really useful e.g. if one plugin used busted and the other one doesn't

**Reply by folke** [reply]
That's all automatic. lazydev doesn't need to know about neoconf.

**Reply by max397574** [reply]
even better
